lang/compiler.go

```diff
@@ -79,3 +114,45 @@ func (ASTCompiler) CompilerType() flux.CompilerType {
 func (c *ASTCompiler) PrependFile(file *ast.File) {
 	c.AST.Files = append([]*ast.File{file}, c.AST.Files...)
 }
+
+// Program implements the flux.Program interface
+type Program struct {
+	deps execute.Dependencies
+	ps   *plan.PlanSpec
+}
+
+func (p Program) Start(ctx context.Context) (flux.Query, error) {
+	e := execute.NewExecutor(p.deps, nil)
+	rmap, _, err := e.Execute(ctx, p.ps, &memory.Allocator{})
+	if err != nil {
+		return nil, err
+	}
+
+	return &Query{
+		rmap: rmap,
+	}, nil
+}
+
+// TODO(cwolff): There should be an easy method presented by the plan package to do this.
```

Comment:
Agreed can you create an issue to track fixing this? And link it to the parent #1085 issue so we track it as part of this whole process?

Reply:
I went ahead and did the work for this:
https://github.com/influxdata/flux/pull/1096
I was planning on merging it to master, and then rebasing this branch when it lands.